### ISSUE TITLE
Support for font-awesome icons?

### ISSUE THREAD

**Issue by Keval17** [dompdf/dompdf#1442]
Does dompdf support font awesome icons?  My html file shows fa fa icons just fine but when I convert it to pdf, nothing comes up. Its totally blank in pdf. 

If is helps, charset in html is set to UTF-8. 

Thanks,
Keval.

**Comment by bsweeney** [comment]
There are some problems rendering with font awesome. In 0.7.0 I was able to get it working by [customizing the font awesome-related CSS](http://stackoverflow.com/a/37289218).

We fixed the specific issue in 0.8.0 but Font Awesome is still broken. I haven't had a chance to identify what's going on in 0.8.0 yet.

**Comment by x00x70** [comment]
I'm not able to get any font-awesome working, using the method posted above? Any idea when this will be fixed in 0.8 ?

**Comment by lexthor** [comment]
any fix for this, yet?

**Comment by akash24** [comment]
yes font awesome working with 0.8.3 but on windows plateform because when i am working with linux system they are not working any help would be appreciated please?

**Comment by netfeld** [comment]
Cannot get even the easyiest example to run under 0.8.3. Perhaps you should post a running example? 

`
require_once 'vendor/autoload.php';

use Dompdf\Dompdf;

$html='
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>TEST DOMPDF</title>
    <link rel="stylesheet" href="font-awesome/css/font-awesome.min.css">   
</head>
<body>
ICON:  <i class="fa fa-check-square-o" aria-hidden="true"></i>
</body>
';

$dompdf = new Dompdf();
$dompdf->loadHtml($html);

$dompdf->setPaper('A4', 'portrait');

// Render the HTML as PDF
$dompdf->render();

// Output the generated PDF to Browser
$dompdf->stream();

`

**Comment by akash24** [comment]
@netfeld is it working ?

**Comment by akash24** [comment]
one more issue regarding dom pdf that I am attaching icons with img src but it's giving server error 500 even I used an absolute path for image icons, anyone can tell why this happening and whats the resolution for this.
note: I am using Ubuntu on my local system instead of windows.

**Comment by bsweeney** [comment]
@akash24 If you copied dompdf from a windows to linux system try removing the generated dompdf_font_family_cache.php file from your font directory. The paths in that file could be wrong for the new system.

As for the 500 error, we'd have to see the PHP error behind that.

**Comment by mycwnet** [comment]
This is also not working for me.  I'm on fedeora, all my css is compiled in sass, font awesome is being imported to the css, the css is working fine but font awesome isn't rendering.  It renders fine when I output the html as a page, but not as a pdf.   Not sure if as someone else suggested this is an issue with linux/apache or something else.

**Comment by tranbaolam6693** [comment]
0.8.3 is still not working for me :( Do I have to config something? CSS is working fine.

**Comment by mycwnet** [comment]
I was able to figure out a way to get this to work for me.  Unfortunately dompdf doesn't seem to be generating the cache file if my font face is local, so I needed to find a remote for fontawesome.   

I'll post my css, but I'd really like to understand how to get a local copy working since I'm not sure the cdn is always going to be reachable.

`@font-face {
font-family: "FontAwesomeRegular";
font-weight: normal;
font-style : normal;
       src : url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/webfonts/fa-regular-400.ttf") format("truetype");
}
@font-face {
font-family: "FontAwesomeBrands";
font-weight: normal;
font-style : normal;
       src : url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/webfonts/fa-brands-400.ttf") format("truetype");
}
@font-face {
font-family: "FontAwesomeSolid";
font-weight: bold;
font-style : normal;
       src : url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.8.2/webfonts/fa-solid-900.ttf") format("truetype");
} 

.fas{
    font-weight: bold !important;
     font-family: FontAwesomeSolid !important;
}
.fas:before{
    font-weight: bold !important;
    font-family: FontAwesomeSolid !important;
}
.fab{
    font-weight: normal !important;
     font-family: FontAwesomeBrands !important;
}
.fab:before{
     font-weight: normal !important;
    font-family: FontAwesomeBrands !important;
}
.far{
    font-weight: normal !important;
     font-family: FontAwesomeRegular !important;
}
.far:before{
     font-weight: normal !important;
    font-family: FontAwesomeRegular !important;
}`

**Comment by bsweeney** [comment]
Would really need to see a sample of the HTML+CSS and how you're using Dompdf before we could provide any helpful feedback.

**Comment by mycwnet** [comment]
My css was be the same as above except the source was pointing to a local ttf, and not a remote one.  Worked fine in html, but not as a pdf.  And when I deleted my dompdf_font_family_cache.php file it wasn't regenerated until I pointed to a remote ttf.

**Comment by bsweeney** [comment]
But what is the path and what is the code that utilizes Dompdf? For non-URL paths the method you load you HTML and the path used both matter.

**Comment by akash24** [comment]
i am trying to include my logo its giving server error while i am including other logo thats coming fine what can be the issue?

**Comment by bsweeney** [comment]
@akash24 your question is unrelated to this issue. Please ask on the support forum http://groups.google.com/group/dompdf

**Comment by mycwnet** [comment]
@bsweeney The path to the local font is ../../../vendor/fortawesome/font-awesome/webfonts/(font name).ttf  not sure if this path would cause problems for dompdf, it seems to be fine for html though.

The code calling dompdf looks like


       $public_dir=$this->getParameter('kernel.project_dir') . '/public/'; 

        $pdfOptions = new Options();
        $pdfOptions->set('defaultFont', 'Arial');
        $pdfOptions->isHtml5ParserEnabled(true);
        $pdfOptions->setFontHeightRatio(1.1);
        $pdfOptions->setTempDir($public_dir . "files/tmp");
        $pdfOptions->isRemoteEnabled(true);
        //$pdfOptions->setDebugCss(true);


        $dompdf = new Dompdf($pdfOptions);

        $html = $this->getResumeHTML();

        $dompdf->loadHtml($html);

        $dompdf->setPaper('A4', 'portrait');

        $dompdf->render();

        $response = new Response($dompdf->output());

        $disposition = HeaderUtils::makeDisposition(
                        HeaderUtils::DISPOSITION_ATTACHMENT,
                        'resume.pdf'
        );

        $response->headers->set('Content-Disposition', $disposition);

        return $response;

**Comment by bsweeney** [comment]
@mycwnet you are loading your HTML via `loadHtml`. This means the document's base domain is the local filesystem and the base path the URL of the script implementing Dompdf. So if you're using a relative URL like that make sure the path specified is valid from when moving through the file system directory tree.

**Comment by dgvai-git** [comment]
any recent fix for font-awesome icons using inside Laravel blade for exporting as pdf? My current version of dompdf is 0.8.5, still not able to get the icons. I gave tried _@mycwnet_'s comment no output!

**Comment by XaibiAslam** [comment]
Anyupdate for this because it is not working still also gradient color is not working as well. Guys any help can be appreciated.

**Comment by bsweeney** [comment]
@XaibiAslam should be working with the latest release. Please submit sample HTML + CSS if you're seeing otherwise.

As for gradients, that's not currently supported. Follow #1912 for updates.
